test/ntttcp_test.py

```diff
@@ -0,0 +1,441 @@
+        file_list = open("file_list.txt", "r").read()
+        filename_pattern = re.compile(r"(.+)\.log")
+        filename = filename_pattern.findall(str(file_list))[0] + ".log"
+        if filename in file_list:
+            return True
+        else:
+            return False
+
+    def parseFile(filename_out):
+        parser = make_parser()
+        parser.setContentHandler(ContentHandler())
+        parser.parse(filename_out)
+
+    def is_output_xml_file(self) -> bool:
+        with open("file_list.txt", "wb") as out:
+            subprocess.run("ls", shell=True, check=True, stdout=out)
+        file_list = open("file_list.txt", "r").read()
+        filename_pattern = re.compile(r"(.+)\.xml")
+        filename = filename_pattern.findall(str(file_list))[0] + ".xml"
+        if filename in file_list:
+            return True
+        else:
+            return False
+        try:
+            parseFile(filename)
+            print('\n\t:), %s is OK!\n' % filename)
+        except Exception as e:
+            return False
+
+    def is_output_json_file(self) -> bool:
+        with open("file_list.txt", "wb") as out:
+            subprocess.run("ls", shell=True, check=True, stdout=out)
+        file_list = open("file_list.txt", "r").read()
+        filename_pattern = re.compile(r"(.+)\.json")
+        filename = filename_pattern.findall(str(file_list))[0] + ".json"
+        filename_out = open(filename, "r")
+        if filename in file_list and json.loads(filename_out.read(), strict=False):
+            return True
+            filename_out.close()
+        else:
+            return False
+
+    #server port starting at:         10000
+    def is_starting_port_number(self, send_cmd) -> bool:
+        starting_port_pattern = re.compile(r"server port starting at:\s+(\d+)")
+        starting_port = starting_port_pattern.findall(str(self.send_stdout))
+        cmd_starting_port_pattern = re.compile(r"-p (\d+)")
+        cmd_starting_port = cmd_starting_port_pattern.findall(send_cmd)
+        if starting_port == cmd_starting_port:
+            return True
+        else:
+            return False
+
+    #ports:                           256
+    def is_mapping_option(self, rec_cmd) -> bool:
+        cmd_mapping_option_pattern = re.compile(r"-m (\d+)")
+        cmd_mapping_option = cmd_mapping_option_pattern.findall(rec_cmd)
+        mapping_option_value_pattern = re.compile(r"ports:\s+(\d+)")
+        mapping_option_value = mapping_option_value_pattern.findall(self.rec_stdout)
+        print("cmd_mapping_option:",cmd_mapping_option)
+        print("mapping_option_value:",mapping_option_value)
+        if cmd_mapping_option[0] == mapping_option_value[0]:
+            return True
+        else:
+            return False
+
+    #show system tcp retransmit:      yes
+    def is_system_tcp_retransmit(self) -> bool:
+        if self.system_tcp_retransmit_pattern.findall(str(self.send_stdout)):
```

Comment:
Some patterns are in the front of the class, some are in the function. We should keep consistent.

Reply:
done